[TECH] Ajouter un test e2e pour un parcours combiné (PIX-19121)

Alexandre-Monney opened the pull request:
## 🔆 Problème
Maintenant que nous avons mis en place le scénario d'un parcours combiné de la reconciliation d'un user jusqu'à la fin de son parcours combiné. On aimerait se proteger pour les futurs développement avec un test e2e retraçant ce scénario

## ⛱️ Proposition
Ajouter un test e2e complet, qui jouera le scénario suivant : 
- Se connecter avec un utilisateur
- Accéder à la page du parcours combiné 
- Rentrer ses infos pour avoir la reconciliation avec l'orga détentrice du parcours
- Jouer la campagne de diagnostique
- Jouer le module recommandé
- Voir l'encart de félicitation en fin de parcours combiné

## 🌊 Remarques
On a remarqué que les tests e2e échouaient systématiquement en cas de retry, car les bases des tests e2e ne sont pas clean entre chaque tests. Ce qui fait qu'il y a des conflits lors des insertions avec contrainte d'unicité.
Donc on passe le retry a 0 dans la config playwright afin de ne pas les faire tourner pour rien

## 🏄 Pour tester
Les tests sont verts

pix-bot-github commented:
Choisir les applications à déployer :

- [ ] [API](https://api-pr13178.review.pix.fr/api/) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-review-pr13178/environment) <!-- pix-api-review -->
- [ ] [App (.fr)](https://app-pr13178.review.pix.fr) – [App (.org)](https://app-pr13178.review.pix.org) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-app-review-pr13178/environment) <!-- pix-app-review -->
- [ ] [Orga (.fr)](https://orga-pr13178.review.pix.fr) – [Orga (.org)](https://orga-pr13178.review.pix.org) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-orga-review-pr13178/environment) <!-- pix-orga-review -->
- [ ] [Certif (.fr)](https://certif-pr13178.review.pix.fr) – [Certif (.org)](https://certif-pr13178.review.pix.org) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-certif-review-pr13178/environment) <!-- pix-certif-review -->
- [ ] [Junior](https://junior-pr13178.review.pix.fr) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-junior-review-pr13178/environment) <!-- pix-junior-review -->
- [ ] [Admin](https://admin-pr13178.review.pix.fr) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-admin-review-pr13178/environment) <!-- pix-admin-review -->
- [ ] [API MaDDo](https://pix-api-maddo-review-pr13178.osc-fr1.scalingo.io/api/) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-maddo-review-pr13178/environment) <!-- pix-api-maddo-review -->
- [ ] [Audit Logger](https://pix-audit-logger-review-pr13178.osc-fr1.scalingo.io/api/) | [👩‍💻 Dashboard Scalingo](https://dashboard.scalingo.com/apps/osc-fr1/pix-audit-logger-review-pr13178/environment) <!-- pix-audit-logger-review -->

> [!IMPORTANT]
> N'oubliez pas de déployer l'API pour pouvoir accéder aux fronts et/ou à l’API MaDDo.

Alexandre-Monney committed:
tests(e2e): add e2e test for combined course scenario

Alexandre-Monney commented on high-level-tests/e2e-playwright/tests/pix-app/combined-course.test.ts:
```diff
@@ -0,0 +1,118 @@
+  await page.getByRole('button', { name: 'Commencer le module' }).click();
+  await page.getByRole('button', { name: 'Terminer' }).click();
+  await page.getByRole('link', { name: 'Continuer' }).click();
+  await page.goto('http://localhost:4200/parcours/COMBINIX1/');
```
**todo**: il n'y a plus besoin de faire ça maintenant qu'on a ajouté le secret la redirection devrait fonctionner automatiquement

aurelie-crouillebois commented on high-level-tests/e2e-playwright/tests/pix-app/combined-course.test.ts:
```diff
@@ -0,0 +1,117 @@
+  await page.getByRole('button', { name: 'Commencer mon parcours' }).click();
+  await page.getByRole('link', { name: 'campagne diag' }).click();
+  await page.getByRole('button', { name: 'Je commence' }).click();
+  await page.getByRole('button', { name: 'Ignorer' }).click();
```
Ajouter commentaires.

aurelie-crouillebois committed:
refactor(e2e): split into function to clarify test

Alexandre-Monney committed:
fix(e2e): set retry count to 0
DB is not cleaned after each test, so we have unicity constraint that crash at first retry

pix-service commented:
:tada: This PR is included in version 5.185.0 :tada:

The release is available on [GitHub release](https://github.com/1024pix/pix/releases/tag/v5.185.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
